riscv-configuration-structure-draft.adoc

```diff
@@ -403,32 +403,47 @@ and then later that same element in a different type describing that same
 component has another value. OPTIONAL elements can be missing in one description
 and be present in another. That is not a conflict.
 
-=== Implementation Examples
-
-There are several options for embedding the binary structure:
-
-1. The structure can describe all harts, and be accessible over the memory bus.
-All harts have the same memory map and the same value in mconfigptr.
-2. There might be several structures in the system, and different harts are
-pointed to different structures by having different pointers in mconfigptr.
-3. There might be several structures in the system. Each hart has the same
-address in mconfigptr. The memory system provides a different configuration
-structure at that address depending on which hart is performing the access.
-4. A combination of 2 and 3 above could be used.
-
-Hardware implementers have a lot of flexibility to handle everything from simple
-fixed systems to complex socketed systems. In each case it's straightforward to
-ensure that each hart can read a Configuration Structure that describes its own
-capabilities.
-
 == Examples
 
 This section gives examples of how the Configuration Structure might be used.
 It is not part of the specification, but should help illustrate the
 specification's intent.
 
+=== Single Structure
+
+Most systems only need a single CS that describes the entire system. Every hart
+has the same value in mconfigptr, and they all access the exact same structure.
+
+=== Multiple Independent Structures
+
+Systems that consist of completely independent parts (e.g. two processors) that
+each run their own code and only communicate at a high level, can have a
+separate CS for each part. Each CS only describes one of the parts, since there
+is no need for any part of the system to have a complete view of the system.
+
+Different harts might be pointed at different CSs by giving them unique
+mconfigptr values. Alternatively, the memory system could provide a different CS
+depending on which hart is accessing a given address. There are many
+possibilities
+
+=== Tree of Structures
+
+When combining IP from multiple vendors it might be helpful to use structure
+pointers (see <<_configuration_structure_pointers>>) instead of combining all
```

Comment:
Please check if this link works or not. It seems problematic here.

Reply:
Hmm... it works in vscode adoc mode, and also in the PDF. But github doesn't seem to recognize this. I'll see if I can change it so it works on github also.

Reply:
Fixed. :-)